### PR TITLE
648 - InternshipEngagement.methodologies should return empty array by default instead of error

### PR DESCRIPTION
**Pull request by zkhin**
#648

### CODE REVIEW

**Comment by zkhin** [comment]
> I know this src code will be refactored soon. Can you add this check to a test (existing or new)?

Okay let me check

**Comment by zkhin** [comment]
@CarsonF I added a new test case for checking that it returns an empty array.

File: test/engagement.e2e-spec.ts [hunk 1]
```diff
@@ -491,6 +497,58 @@ describe('Engagement e2e', () => {
...
+      `,
+      {
+        input: {
+          engagement: internshipEngagementInput,
+        },
+      }
+    );
+
+    const internshipEngagement: InternshipEngagement =
+      result.createInternshipEngagement.engagement;
```
Comment (CarsonF) [inline test/engagement.e2e-spec.ts:531]:
```suggestion
    // Create InternshipEngagement without methodologies
    const internshipEngagement = await createInternshipEngagement(app, {
      projectId: project.id,
      internId: intern.id,
      methodologies: [],
    });
```

Comment (zkhin) [reply test/engagement.e2e-spec.ts]: @CarsonF  Yes, that's a great idea. I changed

**Pull request merged**